webapp/channels/src/components/channel_settings_modal/channel_settings_access_rules_tab.tsx

```diff
@@ -279,27 +297,65 @@ function ChannelSettingsAccessRulesTab({
 
             {/* Auto-sync members toggle */}
             <div className='ChannelSettingsModal__autoSyncSection'>
-                <label className='ChannelSettingsModal__autoSyncLabel'>
+                <label
+                    className='ChannelSettingsModal__autoSyncLabel'
+                    title={(() => {
+                        if (parentPolicyAutoSync === true) {
+                            return formatMessage({
+                                id: 'channel_settings.access_rules.auto_sync_forced_by_parent',
+                                defaultMessage: 'Auto-add is enabled by system policy and cannot be disabled',
+                            });
+                        }
+                        if (!expression.trim()) {
+                            return formatMessage({
+                                id: 'channel_settings.access_rules.auto_sync_requires_expression',
+                                defaultMessage: 'Define access rules to enable auto-add members',
+                            });
+                        }
+                        return undefined;
+                    })()}
+                >
                     <input
                         type='checkbox'
                         className='ChannelSettingsModal__autoSyncCheckbox'
                         checked={autoSyncMembers}
                         onChange={handleAutoSyncToggle}
+                        disabled={(parentPolicyAutoSync === true && autoSyncMembers) || !expression.trim()}
                         id='autoSyncMembersCheckbox'
                         name='autoSyncMembers'
                     />
-                    <span className='ChannelSettingsModal__autoSyncText'>
+                    <span className={`ChannelSettingsModal__autoSyncText ${((parentPolicyAutoSync === true && autoSyncMembers) || !expression.trim()) ? 'disabled' : ''}`}>
                         {formatMessage({
                             id: 'channel_settings.access_rules.auto_sync',
                             defaultMessage: 'Auto-add members based on access rules',
                         })}
                     </span>
                 </label>
                 <p className='ChannelSettingsModal__autoSyncDescription'>
-                    {formatMessage({
-                        id: 'channel_settings.access_rules.auto_sync_description',
-                        defaultMessage: 'Users who match the configured attribute values will be automatically added as members',
-                    })}
+                    {(() => {
+                        if (parentPolicyAutoSync === true) {
```

Comment:
Rather than having logic here in the render block, it will be much cleaner if you moved all the logic outside it, into a useMemo hook and store the result in a const, and use the const here. Having a function here means the function will be called for every single render cycle.

Reply:
Same for the title text of `ChannelSettingsModal__autoSyncLabel`.

Reply:
same, will address it for the next PR